File: backend/routes/feedback.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from db import get_supabase

router = APIRouter()


class FeedbackBody(BaseModel):
    recommendation_id: str
    user_id: str
    rating: int
    feedback_type: str = "helpful"
    comment: str | None = None


@router.post("/feedback")
async def submit_feedback(body: FeedbackBody):
    if body.rating < 1 or body.rating > 5:
        raise HTTPException(status_code=400, detail="Rating must be 1-5")

    sb = get_supabase()
    result = sb.table("feedback_logs").insert({
        "recommendation_id": body.recommendation_id,
        "user_id": body.user_id,
        "rating": body.rating,
        "feedback_type": body.feedback_type,
        "comment": body.comment,
    }).execute()

    return {"success": True, "id": result.data[0]["id"] if result.data else None}

File: src/lib/api/feedback.ts
import type { ApiResponse, RecommendationFeedback } from '@/types';

export async function submitFeedback(
  feedback: RecommendationFeedback
): Promise<ApiResponse<{ success: boolean }>> {
  const res = await fetch('/api/feedback', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      recommendation_id: feedback.recommendationId,
      user_id: '',
      rating: feedback.rating,
      feedback_type: feedback.feedbackType,
      comment: feedback.comment,
    }),
  });

  if (!res.ok) return { data: null, error: 'Failed to submit feedback' };
  const json = await res.json();
  return { data: json, error: null };
}
